server/tests/unit/services/RunService.test.js

```diff
@@ -224,18 +224,421 @@ describe('RunService', () => {
+
+                // Use all available APG Examples for this test
+                const apgExamples = await db.ApgExample.findAll({
+                    where: { test_version_id: testVersion.id }
+                });
+
+                const at = await db.At.findOne({
+                    where: { test_version_id: testVersion.id },
+                    include: [db.AtName]
+                });
+
+                const atVersion = await db.AtVersion.create({
+                    at_name_id: at.AtName.id,
+                    version: atVersionNumber
+                });
+
+                const browser = await db.Browser.findOne();
+                const browserVersion = await db.BrowserVersion.create({
+                    browser_id: browser.id,
+                    version: browserVersionNumber
+                });
+
+                const runStatus = await db.RunStatus.findOne({
+                    where: { name: db.RunStatus.RAW }
+                });
+
+                const user = await db.Users.create();
+                const testCycle = await db.TestCycle.create({
+                    test_version_id: testVersion.id,
+                    created_user_id: user.id
+                });
+
+                let tech = await db.BrowserVersionToAtVersion.create({
+                    at_version_id: atVersion.id,
+                    browser_version_id: browserVersion.id,
+                    active: true,
+                    run_status_id: runStatus.id
+                });
+
+                // Create an active run
+                const previouslyActiveRunEntry = await db.Run.create({
+                    browser_version_id: browserVersion.id,
+                    at_version_id: atVersion.id,
+                    at_id: at.id,
+                    test_cycle_id: testCycle.id,
+                    browser_version_to_at_versions_id: tech.id,
+                    apg_example_id: apgExamples[0].id,
+                    test_version_id: testVersion.id,
+                    active: true,
+                    run_status_id: runStatus.id
+                });
+
+                // Create an inactive run
+                const previouslyInactiveRunEntry = await db.Run.create({
+                    browser_version_id: browserVersion.id,
+                    at_version_id: atVersion.id,
+                    at_id: at.id,
+                    test_cycle_id: testCycle.id,
+                    browser_version_to_at_versions_id: tech.id,
+                    apg_example_id: apgExamples[1].id,
+                    test_version_id: testVersion.id,
+                    active: false,
+                    run_status_id: runStatus.id
+                });
+
+                let previouslyActiveRun = await db.Run.findOne({
+                    where: { id: previouslyActiveRunEntry.id }
+                });
+                let previouslyInactiveRun = await db.Run.findOne({
+                    where: { id: previouslyInactiveRunEntry.id }
+                });
```

Comment:
I'm confused about why you need to `findOne` object you just created. You should have it already if you just created it.

Reply:
Good catch! I honestly have no idea why I did that 😅